commit f8876a79584c0e3e954fa0ea4a5ae4eb97d6c013
Merge: 18fcc36 e06d820
Author: Bernhard Kubicek <kubicek@gmx.at>
Date:   Sun Feb 3 03:05:31 2013 -0800

    Merge pull request #376 from kintel/skip_longfilenames
    
    Also skip .files and _files with long filenames. I think its better than before. If somebody dislikes it, or wants additional configuration, it can be done in a future step. Thanks, Marius, and nice to have you helping!

diff --git a/Marlin/cardreader.cpp b/Marlin/cardreader.cpp
index 9711208..5d9494a 100644
--- a/Marlin/cardreader.cpp
+++ b/Marlin/cardreader.cpp
@@ -90,6 +90,8 @@ void  CardReader::lsDive(const char *prepend,SdFile parent)
     {
       if (p.name[0] == DIR_NAME_FREE) break;
       if (p.name[0] == DIR_NAME_DELETED || p.name[0] == '.'|| p.name[0] == '_') continue;
+      if (longFilename[0] != '\0' &&
+          (longFilename[0] == '.' || longFilename[0] == '_')) continue;
       if ( p.name[0] == '.')
       {
         if ( p.name[1] != '.')
